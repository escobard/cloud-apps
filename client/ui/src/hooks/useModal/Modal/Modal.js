import React from "react";
import PropTypes from "prop-types";
import { Modal, Grid } from "semantic-ui-react";

import "./Modal.scss";

const ModalElement = ({ title, content, showModal, closeModal }) => (
  <Modal id="modal" size="large" open={showModal} onClose={closeModal}>
    <Modal.Content>
      <Grid className="header">
        <Grid.Column width={2}>
          <i
            onKeyDown={e => {
              e.key === "Enter" && closeModal();
            }}
            tabIndex={0}
            aria-label="Back"
            onClick={closeModal}
            className="arrow left big icon"
          />
        </Grid.Column>
        <Grid.Column width={12}>
          <h2>{title}</h2>
        </Grid.Column>
        <Grid.Column width={2}>
          <i aria-hidden="true" className="clipboard outline big icon" />
        </Grid.Column>
      </Grid>
      <div className="icon-container">
        <i
          aria-hidden="true"
          className="sticky note outline big circular icon"
        />
      </div>
      {content}
    </Modal.Content>
  </Modal>
);

Modal.PropTypes = {
  data: PropTypes.shape({
    title: PropTypes.string.isRequired,
    close: PropTypes.object.isRequired
  })
};


export default ModalElement;
