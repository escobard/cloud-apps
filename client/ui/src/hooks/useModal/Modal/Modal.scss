@import "../../../styles/variables";

#modal{
  background: $blue-dark !important;
  > .content{
    background: $blue-dark !important;
    color: $white;
    .ui.message{
      &.red{
        .header {
          color: $red;
        }
      }
      &.green{
        .header {
          color: $green;
        }
      }

    }
    .header{
      color: $white;
      h2{
        text-align: center;
        font-weight: 100;
      }
      .arrow{
        &:hover{
          cursor: pointer;
        }
      }
      .clipboard{
        float: right;
      }
    }
    .icon-container{
      padding: $space-xxlrg 0 $space-mega;
      i{
        font-size: $font-size-xxxl;
        text-align: center;
        float: none;
        margin: auto;
        display: block;
        box-shadow: none;
        border: solid 2px $white;
      }
    }
    button{
      width: 100%;
      text-transform: capitalize;
      font-size: $font-size-small;
      color: $white;
      background: $blue;
      &:hover {
        background: $white;
        color: $blue;
      }
    }
    }
  }
