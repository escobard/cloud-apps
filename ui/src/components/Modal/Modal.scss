@import "../../styles/variables";

div[id*="-modal"]{
  background: $blue-dark !important;
  .content{
    background: $blue-dark !important;
    color: $white;
  }
  h2{
    text-align: center;
    font-weight: 100;
  }
  .clipboard{
    float: right;
  }
}