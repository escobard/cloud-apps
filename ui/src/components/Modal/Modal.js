import React from "react";
import { Modal, Grid } from "semantic-ui-react";

import "./Modal.scss";

const ModalElement = ({ id, open, close, title, content }) => {
  const modalId = `${id}-modal`;

  return (
    <Modal id={modalId} size="large" open={open} onClose={close}>
      <Modal.Content>
        <Grid className="header">
          <Grid.Column width={2}>
            <i
              onKeyDown={e => {
                e.key === "Enter" && close();
              }}
              tabIndex={0}
              aria-label="Back"
              onClick={close}
              className="arrow left big icon"
            />
          </Grid.Column>
          <Grid.Column width={12}>
            <h2>{title}</h2>
          </Grid.Column>
          <Grid.Column width={2}>
            <i aria-hidden="true" className="clipboard outline big icon" />
          </Grid.Column>
        </Grid>
        <div className="icon-container">
          <i
            aria-hidden="true"
            className="sticky note outline big circular icon"
          />
        </div>
        {content}
      </Modal.Content>
    </Modal>
  );
};

export default ModalElement;
