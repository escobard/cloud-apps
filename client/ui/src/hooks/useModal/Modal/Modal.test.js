import Modal from "./index";

describe("Modal component", () => {
  const props = {
    title: <h2>Test title</h2>,
    content: <div>Test content</div>,
    showModal: true,
    closeModal: jest.fn()
  };
  it(">> snapshot is up to date", () => {
    const { container } = render(<Modal {...props} />);
    expect(container).toMatchSnapshot();
  });
  it(">> renders title and content", () => {
    const { getByText } = render(<Modal {...props} />);
    expect(getByText("Test title"));
    expect(getByText("Test content"));
  });
  it(">> triggers closeModal on click", () => {
    const { getByLabelText } = render(<Modal {...props} />);
    fireEvent.click(getByLabelText("Back"));
    expect(props.closeModal).toHaveBeenCalledTimes(1);
  });
  it(">> triggers closeModal on enter key down", () => {
    const { getByLabelText } = render(<Modal {...props} />);
    fireEvent.keyDown(getByLabelText("Back"), {
      key: "Enter",
      code: "Enter",
      keyCode: 13,
      charCode: 13
    });
    expect(props.closeModal).toHaveBeenCalledTimes(1);
  });
});
