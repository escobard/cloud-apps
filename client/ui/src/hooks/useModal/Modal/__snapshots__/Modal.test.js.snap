// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component >> snapshot is up to date 1`] = `<div />`;
